app/controllers/users_controller.rb

```diff
@@ -0,0 +1,34 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```

Comment:
You can just pass in the user_params method you created below.
i.e.
`@user = User.new(user_params)`